### PR TITLE
Games Menu

### PR DESCRIPTION
**Pull request by ManiacVera**
This is the new game menu for Dreamshell

You need:

1.- You need download release 4.0 from Swat page: https://github.com/DC-SWAT/DreamShell/releases/tag/v4.0.0.Release
2.- It just replaces all files in root.
3.- Put 128x128px PNG images in the DS/apps/games/covers folder.
4.- Presets configuration from IsoLoader
5.- Put the games in the root folder Games like RetroDream

Key:
X: Change view
A: Select Game
B: Exit Menu
LT: Previous page
RT: Next page
Pad: Move

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,8 +61,9 @@ SDL_LIBS = -lSDL_$(SDL_VER) \
 IMAGE_LIBS = -lkmg -ljpeg -lpng -lz
 LUA_LIBS = -llua_$(LUA_VER)
 KLIBS = -lkosext2fs -lkosutils -lstdc++ -lm
+TSUNAME_LIBS = -ltsunami_$(TSUNAMI_VER)
```
Comment (DC-SWAT) [inline Makefile:64]: Typo - TSUNAMI

File: applications/games/modules/module.c [hunk 2]
```diff
@@ -0,0 +1,1957 @@
...
+		cmp = strcmp(left_compare, right_compare);
+	}
+
+	free(left_compare);
+	free(right_compare);
+
+	return cmp > 0 ? cmp : -1;
+}
+
+// static int TotalPages()
```
Comment (DC-SWAT) [inline applications/games/modules/module.c:613]: Do not push commented code plz.

**Review by DC-SWAT — COMMENTED** [review]
A couple of suggestions:

1) Call the application differently, just Games somehow doesn’t sound too generic a name, maybe you can come up with a name?
2) If the tsunami library is updated in kos-ports, then here it will remain old (and maybe broken if in KOS changed some API). It would be more correct to use the original one and add C functions for it. Create a header, include the original tsunami header in it and add C functions. Also create a tsunami library for DS, in the name of which you need to add a version so that there are no conflicts or name it as libtsumani-c, and in it you write only the implementation of C functions.

**Comment by ManiacVera** [comment]
> A couple of suggestions:
> 
> 1. Call the application differently, just Games somehow doesn’t sound too generic a name, maybe you can come up with a name?
> 2. If the tsunami library is updated in kos-ports, then here it will remain old (and maybe broken if in KOS changed some API). It would be more correct to use the original one and add C functions for it. Create a header, include the original tsunami header in it and add C functions. Also create a tsunami library for DS, in the name of which you need to add a version so that there are no conflicts or name it as libtsumani-c, and in it you write only the implementation of C functions.


1.- I really don't have another name, but if you suggest one we can use it, the idea is not to waste all the work I did.
2.- The whole tsunami library changed, I had to remove all smart pointers, if you compare the original code it is different. I put all the files in a different folder than the original one to prevent it from failing in future versions (it is not updateable with kos-ports), I can call it tsunami_ds, so it would be an exclusive version for dreamshell.

My friend, it was 3 months of work, don't let this get lost.

Thanks!

File: applications/games/modules/module.c (hunk 2 above)
Comment (ManiacVera) [reply applications/games/modules/module.c:613]: I didn't see that I left commented code, can I delete it, do I make the adjustment and upload it? Or what should I do?

**Comment by DC-SWAT** [comment]
> > A couple of suggestions:
> > 
> > 1. Call the application differently, just Games somehow doesn’t sound too generic a name, maybe you can come up with a name?
> > 2. If the tsunami library is updated in kos-ports, then here it will remain old (and maybe broken if in KOS changed some API). It would be more correct to use the original one and add C functions for it. Create a header, include the original tsunami header in it and add C functions. Also create a tsunami library for DS, in the name of which you need to add a version so that there are no conflicts or name it as libtsumani-c, and in it you write only the implementation of C functions.
> 
> 1.- I really don't have another name, but if you suggest one we can use it, the idea is not to waste all the work I did. 2.- The whole tsunami library changed, I had to remove all smart pointers, if you compare the original code it is different. I put all the files in a different folder than the original one to prevent it from failing in future versions (it is not updateable with kos-ports), I can call it tsunami_ds, so it would be an exclusive version for dreamshell.
> 
> My friend, it was 3 months of work, don't let this get lost.
> 
> Thanks!

1) You can call it that Games Menu.
2) I see, OK. Let's leave it like that.

Good job, just make few changes and I merge it!

**Comment by ManiacVera** [comment]
> > > A couple of suggestions:
> > > 
> > > 1. Call the application differently, just Games somehow doesn’t sound too generic a name, maybe you can come up with a name?
> > > 2. If the tsunami library is updated in kos-ports, then here it will remain old (and maybe broken if in KOS changed some API). It would be more correct to use the original one and add C functions for it. Create a header, include the original tsunami header in it and add C functions. Also create a tsunami library for DS, in the name of which you need to add a version so that there are no conflicts or name it as libtsumani-c, and in it you write only the implementation of C functions.
> > 
> > 
> > 1.- I really don't have another name, but if you suggest one we can use it, the idea is not to waste all the work I did. 2.- The whole tsunami library changed, I had to remove all smart pointers, if you compare the original code it is different. I put all the files in a different folder than the original one to prevent it from failing in future versions (it is not updateable with kos-ports), I can call it tsunami_ds, so it would be an exclusive version for dreamshell.
> > My friend, it was 3 months of work, don't let this get lost.
> > Thanks!
> 
> 1. You can call it that Games Menu.
> 2. I see, OK. Let's leave it like that.
> 
> Good job, just make few changes and I merge it!

1.- Ok perfect "Games Menu".
2.- Thank you for understanding it was a great effort, Even I dream about the code, it was a nightmare to remove all the smart pointers, to be able to make it work.

Tomorrow I will make the adjustments to upload them.
Thanks Swat

File: applications/games/modules/module.c (hunk 2 above)
Comment (DC-SWAT) [reply applications/games/modules/module.c:613]: > I didn't see that I left commented code, can I delete it, do I make the adjustment and upload it? Or what should I do?

Yes, of course, continue to push to your repository in the same way and the changes will be automatic here too. This is the power of Github.

Comment (ManiacVera) [reply applications/games/modules/module.c]: Hi, I already uploaded them, did you get the modified files?

Comment (DC-SWAT) [reply applications/games/modules/module.c]: Yes! But you forgot to fix type in    root Makefile.

File: Makefile (hunk 1 above)
Comment (DC-SWAT) [reply Makefile:64]: Rename TSUNAME to TSUNAMI.

Comment (ManiacVera) [reply Makefile:64]: Yes, i forgot

Comment (ManiacVera) [reply Makefile]: I already renamed it

**Pull request merged**